aws/validators.go

```diff
@@ -124,9 +124,9 @@ func validateNeptuneEngine() schema.SchemaValidateFunc {
 
 func validateElastiCacheClusterId(v interface{}, k string) (ws []string, errors []error) {
 	value := v.(string)
-	if (len(value) < 1) || (len(value) > 20) {
+	if (len(value) < 1) || (len(value) > 30) {
```

Comment:
where have you seen this, the documentation does list the max as 20.
https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/GettingStarted.CreateCluster.html
see: 
"Cluster naming constraints" under Step 6

Reply:
Yeah, I saw that in the AWS docs too. I observed the issue with infra created by a Terraform config and confirmed this addressed the problem.